src/components/preferences/PreferencesPage.js

```diff
@@ -0,0 +1,87 @@
+        actions.setSubmitting = true;
+        console.log(values);
+    };
+
+    const restoreDefaults = (setFieldValue) => (event) => {
+        setFieldValue("preferences.rememberLastPath", true);
+        setFieldValue("preferences.saveSession", true);
+        setFieldValue("preferences.enableImportEmailNotification", true);
+        setFieldValue("preferences.enableWaitTimeMessage", true);
+        setFieldValue("preferences.enableAnalysisEmailNotification", true);
+        setFieldValue("preferences.enableHPCPrompt", true);
+        setFieldValue("preferences.notificationKBShortcut", "N");
+        setFieldValue("preferences.dataKBShortcut", "D");
+        setFieldValue("preferences.closeKBShortcut", "Q");
+        setFieldValue("preferences.appsKBShortcut", "A");
+        setFieldValue("preferences.analysisKBShortcut", "Y");
+        // TODO find a way to get users default output folder
+        //setFieldValue("preferences.default_output_folder", )
+        setRestoreDef(false);
+    };
+
+    return (
+        <div className={classes.root}>
+            <Formik initialValues={config} onSubmit={handleSubmit}>
+                {(props) => (
+                    <Form>
+                        <PreferencesTab config={props} />
+                        <br />
+                        <br />
+                        <ShortcutsTab />
+                        <br />
+                        <DialogActions>
+                            <Button
+                                className={classes.actionButton}
+                                color="primary"
+                                onClick={() => setRestoreDef(true)}
+                            >
+                                RESTORE DEFAULTS
+                            </Button>
+                            <Button
+                                className={classes.actionButton}
+                                color="primary"
+                                type="submit"
+                            >
+                                SAVE
+                            </Button>
+                        </DialogActions>
+                        <DEConfirmationDialog
+                            dialogOpen={restoreDef}
+                            debugID="temp"
+                            onOkBtnClick={restoreDefaults(props.setFieldValue)}
+                            okLabel="Restore Defaults"
+                            onCancelBtnClick={() => setRestoreDef(false)}
+                            heading="Restore Defaults"
+                            messages="messaging"
+                            message={
+                                <LoadingMask loading={false}>
+                                    <div>
+                                        Are you sure you would like to restore
+                                        defaults?
+                                    </div>
+                                </LoadingMask>
```

Comment:
I mentioned this on our call today in regards to something else, but we're kind of shifting away from the loading spinners to the [skeletons](https://material-ui.com/components/skeleton/). I don't think it's necessarily bad to have a spinner on a dialog though.  The alternative would be to close the dialog and apply the skeleton to the preferences page while the preferences are saving.  I don't feel strongly one way or the other, just noting it here.